Src/POI.DiscordDotNet/Services/Implementations/DiscordStarboardService.cs

```diff
@@ -0,0 +1,128 @@
+
+	public DiscordStarboardService(
+		ILogger<DiscordChatCommandsService> logger,
+		IStarboardMessagesRepository starboardMessagesRepository, IServerSettingsRepository serverSettingsRepository)
+	{
+		_logger = logger;
+		_starboardMessagesRepository = starboardMessagesRepository;
+		_serverSettingsRepository = serverSettingsRepository;
+	}
+
+	public void Setup(IDiscordClientProvider discordClientProvider)
+	{
+		_logger.LogDebug("Setting up DiscordStarboardService");
+
+		var client = discordClientProvider.Client!;
+		client.MessageReactionAdded -= OnReactionAdded;
+		client.MessageReactionAdded += OnReactionAdded;
+	}
+
+	public void Cleanup(IDiscordClientProvider discordClientProvider)
+	{
+		_logger.LogDebug("Cleaning up DiscordStarboardService");
+		var client = discordClientProvider.Client!;
+		client.MessageReactionAdded -= OnReactionAdded;
+	}
+
+	private async Task OnReactionAdded(DiscordClient sender, MessageReactionAddEventArgs args)
+	{
+		//TODO: Check if message was not cached
+		//TODO: Check on emoji ID, not name
+		//TODO: Figure out a way to get the display name of the user outside the guild...
+
+		var serverSettings = await _serverSettingsRepository.FindOneById(args.Channel.Guild.Id);
+		if (serverSettings?.StarboardChannelId == null)
+		{
+			_logger.LogError("Server settings or starboard channel id not found!");
+			return;
+		}
+
+		// Skip event if the message is in the starboard channel (To prevent people staring the bot messages)
+		if(args.Channel.Id == serverSettings.StarboardChannelId)
+		{
+			return;
+		}
+
+		// Check if the message reactions contains the star emote
+		if (args.Message.Reactions.All(x => x.Emoji.Name != "⭐"))
+		{
+			return;
+		}
+
+		var messageStarCount = args.Message.Reactions.First(x => x.Emoji.Name == "⭐").Count;
+		// Check if the message has enough stars
+		if (messageStarCount < serverSettings.StarboardEmojiCount)
+		{
+			return;
+		}
+
+		// Get the starboard channel by the server settings id
+		var starboardChannel = await sender.GetChannelAsync(serverSettings.StarboardChannelId.Value);
+		if (starboardChannel == null)
+		{
+			_logger.LogError("Starboard channel not found!");
+			return;
+		}
+
+		// Get DiscordMember from DiscordUser to get the display name
+		var author = await args.Guild.GetMemberAsync(args.Message.Author.Id);
+		if (author == null)
+		{
+			_logger.LogError("Author not found!");
+			return;
+		}
+
+		// Get the starboard message from the database
+		var foundMessage = await _starboardMessagesRepository.FindOneByServerIdAndChannelIdAndMessageId(args.Guild.Id, args.Channel.Id, args.Message.Id);
+		// If the message is already in the database, update the star count
+		// (This will also update the message contents)
+		var embed = GetStarboardEmbed(author.DisplayName, args.Message.Channel.Name, args.Message.Content, args.Message.JumpLink, args.Message.Timestamp, (uint)messageStarCount, args.Message.Attachments.FirstOrDefault()?.Url);
+
+		if(foundMessage != null)
```

Comment:
Preferably also invert the if so it's `if (foundMessage == null)` and use an else block instead of return